src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
While this is a rather serious problem, I'm going to merge this PR unless someone tells me not to because current mending system is a common source of legitimate complaints and frustration.

This regression should be fixed ASAP and should be treated like a bug, though.

> As the activity isn't suspendable it won't get written out to the save

Not true: an autosave in the middle of the activity will write the activity with hardcoded switch values. Future updates will need to handle that.